sesman/chansrv/chansrv_fuse.c

```diff
@@ -738,11 +762,17 @@ int xfuse_create_share(tui32 device_id, const char *dirname)
 {
     int result = -1;
     XFS_INODE  *xinode;
+    size_t dirnamelen = strlen(dirname);
 
-    if (dirname != NULL && strlen(dirname) > 0 &&
+    if (dirname != NULL && dirnamelen > 0 &&
             xfuse_init_xrdp_fs() == 0)
     {
-        xinode = xfs_add_entry(g_xfs, FUSE_ROOT_ID, dirname, (0777 | S_IFDIR));
+        char *newdirname = _fuse_mount_name_colon_char_replace(dirname, dirnamelen);
+        xinode = xfs_add_entry(g_xfs, FUSE_ROOT_ID, newdirname, (0777 | S_IFDIR));
+        if (newdirname != dirname)
```

Comment:
Nice way to add the free! Suggest a comment somewhere to explain this a bit more.